freqtrade/exchange/__init__.py

```diff
@@ -8,52 +8,43 @@
 from cachetools import cached, TTLCache
 from datetime import datetime
 
+import ccxt
 import arrow
-import requests
+from cachetools import cached, TTLCache
+
+from freqtrade import OperationalException, DependencyException, NetworkException
 
-from freqtrade import OperationalException, NetworkException
 
 logger = logging.getLogger(__name__)
 
 # Current selected exchange
-_API = None
+_API: ccxt.Exchange = None
+
 _CONF: dict = {}
 API_RETRY_COUNT = 4
 
 # Holds all open sell orders for dry_run
 _DRY_RUN_OPEN_ORDERS: Dict[str, Any] = {}
 
+_TICKER_CACHE: dict = {}
+
 def retrier(f):
     def wrapper(*args, **kwargs):
         count = kwargs.pop('count', API_RETRY_COUNT)
         try:
             return f(*args, **kwargs)
-        # TODO dont be a gotta-catch-them-all pokemon collector
-        except Exception as ex:
-            logger.warn('%s returned exception: "%s"', f, ex)
+        except (NetworkException, DependencyException) as ex:
+            logger.warning('%s returned exception: "%s"', f, ex)
             if count > 0:
                 count -= 1
                 kwargs.update({'count': count})
-                logger.warn('retrying %s still for %s times', f, count)
+                logger.warning('retrying %s still for %s times', f, count)
                 return wrapper(*args, **kwargs)
             else:
                 raise OperationalException('Giving up retrying: %s', f)
     return wrapper
 
 
-def _get_market_url(exchange):
```

Comment:
I don't see why to remove this? We use the exchange pair-links in the telegram

Reply:
@shusso As far as I can see this has been replaced with the `_EXCHANGE_URLS` mapping